src/window/src/lib.rs

```diff
@@ -1,18 +1,32 @@
 #![deny(missing_docs)]
 #![deny(missing_copy_implementations)]
 
-//! Window abstraction
+//! Window storage and interfacing traits.
+//!
+//! The Window, AdvancedWindow, BuildFromWindowSettings traits are used by
+//! Piston to interface with its many backends. They aren't generally useful
+//! for the average piston user.
```

Comment:
The `Window` and `AdvancedWindow` traits are used by generic code that does not depend on a specific window backend:

- `Window` has the minimum requirements to get the default game-loop working
- `AdvancedWindow` inherits from `Window` and has full window functionality

When you write generic code, it is common to use the trait with least requirements to increase the portability of the code.